### PR TITLE
PassThrough Auth Docs

### PR DESCRIPTION
**Pull request by sam-heilbron**
# Description

Update the PassThrough Auth docs to include instructions for sharing state with other plugins

# Context

PassThrough Auth now behaves similarly to plugin auth (https://docs.solo.io/gloo-edge/latest/guides/dev/writing_auth_plugins/#sharing-state-between-steps) in that it can share state with other plugins.

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: docs/content/guides/security/auth/extauth/passthrough_auth/_index.md [hunk 1]
```diff
@@ -203,7 +203,26 @@ The sample gRPC authentication service has been implemented such that any reques
...
-The request should now be authorized! 
+The request should now be authorized!
+
+## Sharing state with other plugins
+
+{{% notice note %}}
+The sharing state feature was introduced with **Gloo Edge Enterprise**, release 1.6.10. If you are using an earlier version, this will not work.
+{{% /notice %}}
+
+A common requirement is to be able to share state between the Passthrough service, and other auth plugins. When writing a custom auth plugin, this is possible, and the steps to achieve it are [outlined here]({{< versioned_link_path fromRoot="/guides/dev/writing_auth_plugins#sharing-state-between-steps" >}}). We support this requirement by leveraging request and response metadata.
```
Comment (kdorosh) [inline docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:214]: It's not just between auth plugins, but also our built in auth.

The most common flows I've seen are users using the JWT ID token from the OIDC flow and using that in custom logic

File: docs/content/guides/security/auth/extauth/passthrough_auth/_index.md [hunk 2]
```diff
@@ -203,7 +203,26 @@ The sample gRPC authentication service has been implemented such that any reques
...
+
+We provide some example implementations in the Gloo Edge repository at `docs/examples/passthrough-auth/pkg/auth/v3/auth-with-state.go`.
+
+### Reading state from other plugins
```
Comment (kdorosh) [inline docs/content/guides/security/auth/extauth/passthrough_auth/_index.md:218]: Let's avoid the word plugins when it really isn't a plug-in

File: docs/examples/grpc-passthrough-auth/pkg/auth/v3/auth-with-state.go [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	if filterMetadata == nil {
+		log.Println("Request does not have FilterMetadata")
+		return unauthorizedResponse()
+	}
+
+	// This is the state that is made available to this passthrough service
+	availablePassThroughState := filterMetadata[soloPassThroughAuthMetadataKey]
+
+	var jwt string
+	if jwtFromState, ok := availablePassThroughState.GetFields()["jwt"]; ok {
```
Comment (kdorosh) [inline docs/examples/grpc-passthrough-auth/pkg/auth/v3/auth-with-state.go:42]: Add a comment to say where this is supposed to be coming from (who auth sets the JWT?)

**Review by kdorosh — COMMENTED** [review]
Some minor suggestions

**Review by saiskee — APPROVED** [review]
Looks good to me! Addressed all of kevin's comments and couldn't find anything else needed to address.

**Pull request merged**